### DISCUSSION TITLE
System.MissingMethodException: Method not found: double Microsoft.UI.Xaml.Thickness.get_Left()

### DISCUSSION THREAD

**Discussion by giraudremy** [CommunityToolkit/Windows#278] [Q&A]
Hi,

I'm using CommunityToolkit with uno platform and in my Wasm project I have this exception :

> fail: Uno.UI.Dispatching.NativeDispatcher[0] NativeDispatcher unhandled exception System.MissingMethodException: Method not found: double Microsoft.UI.Xaml.Thickness.get_Left() at Microsoft.UI.Xaml.FrameworkElement.InnerMeasureCore(Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs:line 103 at Microsoft.UI.Xaml.FrameworkElement.MeasureCore(Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs:line 81 at Microsoft.UI.Xaml.UIElement.Measure(Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/UIElement.Layout.crossruntime.cs:line 176 at Microsoft.UI.Xaml.FrameworkElement.MeasureElement(UIElement view, Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/FrameworkElement.cs:line 301 at Microsoft.UI.Xaml.Controls.ItemsPresenter.MeasureOverride(Size size) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs:line 459 at Microsoft.UI.Xaml.FrameworkElement.InnerMeasureCore(Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs:line 103 at Microsoft.UI.Xaml.FrameworkElement.MeasureCore(Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/FrameworkElement.Layout.crossruntime.cs:line 81 at Microsoft.UI.Xaml.UIElement.Measure(Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/UIElement.Layout.crossruntime.cs:line 176 at Microsoft.UI.Xaml.FrameworkElement.MeasureElement(UIElement view, Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/FrameworkElement.cs:line 301 at Microsoft.UI.Xaml.FrameworkElement.MeasureFirstChild(Size availableSize) in /agent/_work/1/s/src/Uno.UI/UI/Xaml/FrameworkElement.cs:line 747 

I'm previouly using Uno.CommunityTollkit.WinUI.UI.*** 7.1.100. T**o fix this issue I have update to the last pre release.**

I have now migrate to CommunityTollkit.WinUI 8.0 and I refound this issue but I have not found pre release...

Is there a new version or a new pre-release of v8.0 planned?

Thank for your help.

**Comment by michael-hawker** [comment]
@giraudremy can you confirm what version of Uno you are using in your project?

**Reply by giraudremy** [reply]
I'am using Uno 5.0 and I have also try 5.1 pre release (5.1.0-dev.542).

**Reply by giraudremy** [reply]
Hi, 

I just migrated to the latest versions of platformuno (5.1.104) and Communitytoolkit (8.0.240109) and I still have the same error. I thought it had been fixed via “https://github.com/CommunityToolkit/Tooling-Windows-Submodule/pull/157”. Is this really the case? Which version should I use?

Thanks

**Reply by michael-hawker** [reply]
@giraudremy can you also clarify what exactly you're trying to use from the Toolkit? (The stack above is all just Uno code.)

8.0.240109 doesn't support Uno 5, but we did just ship [a pre-release of `8.1.240328-rc`](https://github.com/CommunityToolkit/Windows/releases/tag/v8.1.240328-rc) that supports Uno 5.1.87+.

If that doesn't work, please open an issue. It'd be great to get a stand-alone minimal repro with just the code that's failing for you. That would be handy for us to investigate within the toolkit or be able to hand to the Uno team if we think it's an issue specific to Uno.

We also just updated https://toolkitlabs.dev/ with the latest code, so if you let us know what component you're using, we can also see if it reproduces there.

**Reply by giraudremy** [reply]
Hello,

I am using a wrap panel like this:

```
<GridView.ItemsPanel>
    <ItemsPanelTemplate>
        <controlsToolkit:WrapPanel Orientation="Horizontal" HorizontalAlignment="Center" />
    </ItemsPanelTemplate>
</GridView.ItemsPanel>
<GridView.ItemTemplate>
```

When I use the CommunityToolkit.WinUI.UI.Controls" Version=“7.1.2” for the WinUi3 part and the Uno version for the Wasm part, I have no errors, everything works fine. 

With version 8.0.240109, for the WinUi3 part I have no errors and for the wasm part I have the error: Unhandled exception System.MissingMethodException: Method not found: double Microsoft.UI.Xaml.Thickness.get_Left()

And with version 8.1.240328-rc, I was not able to compile the WinUi3 project and I have the error: Xaml Internal Error error WMC9999: Type universe cannot resolve assembly: Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null.

Even though in the WinUi3 project I do not reference Uno at all and I use the WinUi version and not Uwp, so I do not understand this new dependency.

So, I went back to version 7.1.2.
